lib/src/widget/asset_picker.dart

```diff
@@ -67,14 +66,18 @@ class AssetPicker<A, P> extends StatelessWidget {
       requestType = RequestType.common;
     }
     if ((specialItemBuilder == null &&
-            specialItemPosition != SpecialItemPosition.none) ||
+        specialItemPosition != SpecialItemPosition.none) ||
         (specialItemBuilder != null &&
             specialItemPosition == SpecialItemPosition.none)) {
       throw ArgumentError('Custom item didn\'t set properly.');
     }
 
     try {
-      final bool isPermissionGranted = await PhotoManager.requestPermission();
+      final bool isPermissionGranted =
+          (await PhotoManager.requestPermissionExtend() ==
+              PermissionState.authorized) ||
+              (await PhotoManager.requestPermissionExtend() ==
+                  PermissionState.limited);
       if (isPermissionGranted) {
```

Comment:
Don't use duplicate await.

```suggestion
      final PermissionState _ps = await PhotoManager.requestPermissionExtend();
      if (_ps == PermissionState.authorized || _ps == PermissionState.limited) {
```